### PR TITLE
Docs: HLRC: refactor bulk, migrate and reindex apis

### LINKED ISSUE #35345: HLRC: remove duplication from ascii docs
there is a way to decrease duplication when writing the documentation for HLRC

Things to fix:
- migration API - once commited #34898
- bulk API - mostly done by @nik9000 in #34925, small improvements in #35413 
- reindex API - improvments in #35413

see #34925

### PR DESCRIPTION
**Pull request by pgomulka**
fix typos and refactor to DRY up documentation for bulk, reindex and migration apis
relates #35345

### CODE REVIEW

**Comment by pgomulka** [comment]
this is work in progress, will add more from the migration API once commited

File: docs/java-rest/high-level/document/bulk.asciidoc [hunk 1]
```diff
@@ -37,9 +37,9 @@ And different operation types can be added to the same +{request}+:
 --------------------------------------------------
 include-tagged::{doc-tests-file}[{api}-request-with-mixed-operations]
 --------------------------------------------------
-<1> Adds a `DeleteRequest` to the `BulkRequest`. See <<{upid}-delete>>
+<1> Adds a `DeleteRequest` to the +{request}+. See <<{upid}-delete>>
 for more information on how to build `DeleteRequest`.
-<2> Adds an `UpdateRequest` to the `BulkRequest`. See <<{upid}-update>>
+<2> Adds an `UpdateRequest` to the +{request}+. See <<{upid}-update>>
```
Comment (nik9000) [inline docs/java-rest/high-level/document/bulk.asciidoc]: :heart:

**Review by nik9000 — APPROVED** [review]
Nice! I think it is an improvement.

**Comment by pgomulka** [comment]
@elasticmachine ok to test

**Comment by jakelandis** [comment]
@pgomulka - It looks like this was back ported on https://github.com/elastic/elasticsearch/commit/7df652ab4c9977813fee73b5d4abd0af4d9674fd ?  If so, can you please remove the `backport pending` label.

**Comment by pgomulka** [comment]
sure - i totally missed that. thank you for spotting

**Pull request merged**